onadata/libs/utils/export_tools.py

```diff
@@ -692,7 +692,10 @@ def clean_keys_of_slashes(record):
 
 
 def _get_server_from_metadata(xform, meta, token):
-    report_templates = MetaData.external_export(xform)
+    from multidb.pinning import use_master
+    # Fetch metadata details from master directly
+    with use_master:
+        report_templates = MetaData.external_export(xform)
 
```

Comment:
In other parts of the codebase, we have taken the approach of using master only when the initial request has failed. I would see something like:
```python
try:
    report_templates = MetaData.external_export(xform)
except MetaData.DoesNotExist:
   with use_master:
        report_templates = MetaData.external_export(xform)
```